File: app/src/main/java/kr/co/yogiyo/rookiephotoapp/diary/main/DiariesAdapter.java
package kr.co.yogiyo.rookiephotoapp.diary.main;

import android.content.Context;
import android.content.Intent;
import android.os.Environment;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;

import java.io.File;
import java.util.Calendar;
import java.util.Date;
import java.util.GregorianCalendar;
import java.util.List;

import kr.co.yogiyo.rookiephotoapp.R;
import kr.co.yogiyo.rookiephotoapp.diary.DiaryDetailActivity;
import kr.co.yogiyo.rookiephotoapp.diary.db.Diary;

public class DiariesAdapter extends RecyclerView.Adapter<DiariesAdapter.DiariesViewHolder> {

    private Context context;
    private List<Diary> diaries;

    public DiariesAdapter(Context context, List<Diary> diaries) {
        this.context = context;
        this.diaries = diaries;
    }

    @NonNull
    @Override
    public DiariesViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.item_diary, viewGroup, false);

        return new DiariesViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull DiariesViewHolder diariesViewHolder, int i) {
        Diary diary = diaries.get(i);
        Date date = diary.getDate();
        Calendar calendar = new GregorianCalendar();
        calendar.setTime(date);

        diariesViewHolder.dayText.setText(String.format(context.getString(R.string.text_day_now_diary), calendar.get(Calendar.DAY_OF_MONTH)));
        diariesViewHolder.timeText.setText(String.format(context.getString(R.string.text_time_now_diary),
                calendar.get(Calendar.HOUR_OF_DAY),
                calendar.get(Calendar.MINUTE) < 10 ?
                        "0" + calendar.get(Calendar.MINUTE) : String.valueOf(calendar.get(Calendar.MINUTE))));
        diariesViewHolder.descriptionText.setText(diary.getDescription());

        File yogiDiaryStorageDir = Environment.getExternalStoragePublicDirectory(String.format(context.getString(R.string.text_yogidiary_path),
                Environment.DIRECTORY_PICTURES, "YogiDiary"));

        String imageAbsolutePath = String.format("%s/%s", yogiDiaryStorageDir.getAbsolutePath(), diary.getImage());
        Glide.with(context)
                .load(imageAbsolutePath)
                .diskCacheStrategy(DiskCacheStrategy.NONE)
                .skipMemoryCache(true)
                .into(diariesViewHolder.imageView);
    }

    @Override
    public int getItemCount() {
        return diaries.size();
    }

    public void setItems(List<Diary> diaries) {
        this.diaries = diaries;
    }

    public class DiariesViewHolder extends RecyclerView.ViewHolder {

        private TextView dayText;
        private TextView timeText;
        private TextView descriptionText;
        private ImageView imageView;

        // TODO: View 클릭시 해당 다이어리의 상세 화면으로 이동
        public DiariesViewHolder(@NonNull View itemView) {
            super(itemView);
            dayText = itemView.findViewById(R.id.text_day);
            timeText = itemView.findViewById(R.id.text_time);
            descriptionText = itemView.findViewById(R.id.text_description);
            imageView = itemView.findViewById(R.id.image_view);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    Log.d(DiariesAdapter.class.getSimpleName(), diaries.get(getAdapterPosition()).getIdx() + "");
                    Intent intent = new Intent(context, DiaryDetailActivity.class);
                    intent.putExtra("DIARY_IDX", diaries.get(getAdapterPosition()).getIdx());
                    context.startActivity(intent);
                }
            });
        }
    }
}

File: app/src/main/java/kr/co/yogiyo/rookiephotoapp/diary/main/DiariesAdapter.kt
package kr.co.yogiyo.rookiephotoapp.diary.main

import android.content.Context
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import kr.co.yogiyo.rookiephotoapp.Constants
import kr.co.yogiyo.rookiephotoapp.R
import kr.co.yogiyo.rookiephotoapp.diary.DiaryDetailActivity
import kr.co.yogiyo.rookiephotoapp.diary.db.Diary
import java.text.SimpleDateFormat
import java.util.*

class DiariesAdapter(val context: Context, var diaries: List<Diary>) : RecyclerView.Adapter<DiariesAdapter.DiariesViewHolder>() {

    private var hourMinuteFormat = SimpleDateFormat("hh:mm", Locale.getDefault())
    private var dayFormat = SimpleDateFormat("dd일", Locale.getDefault())

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): DiariesViewHolder {
        val view = LayoutInflater.from(viewGroup.context)
                .inflate(R.layout.item_diary, viewGroup, false)

        return DiariesViewHolder(view)
    }

    override fun onBindViewHolder(diariesViewHolder: DiariesViewHolder, i: Int) {

        val diary = diaries[i]
        val date = diary.date
        val calendar = GregorianCalendar().apply {
            time = date
        }

        diariesViewHolder.dayText.text = dayFormat.format(calendar.time)
        diariesViewHolder.timeText.text = hourMinuteFormat.format(calendar.time)
        diariesViewHolder.descriptionText.text = diary.description

        val yogiDiaryStorageDir = Constants.YOGIDIARY_PATH

        val imageAbsolutePath = "${yogiDiaryStorageDir.absolutePath}/${diary.image}"
        Glide.with(context)
                .load(imageAbsolutePath)
                .diskCacheStrategy(DiskCacheStrategy.NONE)
                .error(R.drawable.baseline_not_interested_black_36)
                .into(diariesViewHolder.imageView)
    }

    override fun getItemCount(): Int {
        return diaries.size
    }

    fun setItems(diaries: List<Diary>) {
        this.diaries = diaries
    }

    inner class DiariesViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        val dayText: TextView = itemView.findViewById(R.id.text_day)
        val timeText: TextView = itemView.findViewById(R.id.text_time)
        val descriptionText: TextView = itemView.findViewById(R.id.text_description)
        val imageView: ImageView = itemView.findViewById(R.id.image_view)

        init {
            itemView.setOnClickListener {
                val intent = Intent(context, DiaryDetailActivity::class.java)
                intent.putExtra("DIARY_IDX", diaries[adapterPosition].idx)
                context.startActivity(intent)
            }
        }
    }
}
